python/pyspark/mllib/linalg.py

```diff
@@ -1152,9 +1156,385 @@ def sparse(numRows, numCols, colPtrs, rowIndices, values):
         return SparseMatrix(numRows, numCols, colPtrs, rowIndices, values)
 
 
+class DistributedMatrix(object):
+    """
+    Represents a distributively stored matrix backed by one or
+    more RDDs.
+
+    """
+    def numRows(self):
+        """Get or compute the number of rows."""
+        raise NotImplementedError
+
+    def numCols(self):
+        """Get or compute the number of cols."""
+        raise NotImplementedError
+
+
+class RowMatrix(DistributedMatrix):
+    """
+    Represents a row-oriented distributed Matrix with no meaningful
+    row indices.
+
+    .. note:: Experimental
+    """
+    def __init__(self, rows, numRows=0, numCols=0):
+        """
+        Create a wrapper over a Java RowMatrix.
+
+        :param rows: An RDD of Vectors.
+        """
+        if not isinstance(rows, RDD):
+            raise TypeError("rows should be an RDD object, got %s" % type(rows))
+        first = rows.first()
+        if not isinstance(first, Vector):
+            raise TypeError("rows should be an RDD of Vectors, got an RDD of %s" % type(first))
+
+        javaRowMatrix = callMLlibFunc("createRowMatrix", rows, long(numRows), int(numCols))
+        self._jrm = JavaModelWrapper(javaRowMatrix)
+        self.rows = rows
+
+    @staticmethod
+    def _from_java(javaRowMatrix):
+        """Create a PySpark RowMatrix from a Java RowMatrix object."""
+        rows = JavaModelWrapper(javaRowMatrix).call("rows")
+        return RowMatrix(rows)
+
+    def numRows(self):
+        """
+        Get or compute the number of rows.
+
+        >>> rows = sc.parallelize([Vectors.dense([1, 2, 3]), Vectors.dense([4, 5, 6]),
+        ...                        Vectors.dense([7, 8, 9]), Vectors.dense([10, 11, 12])])
+        >>> rm = RowMatrix(rows)
+        >>> int(rm.numRows())
+        4
+
+        >>> rows = sc.parallelize([Vectors.dense([1, 2, 3]), Vectors.dense([4, 5, 6]),
```

Comment:
nit: I think this can be removed, since `rows` is unchanged? If this be removed, it will also be clearer that the `numRows` and `numCols` depends on the parameters supplied rather than the actual size of RDD's for the same input.